### PR TITLE
Destination Monetate

### PR DESCRIPTION
**Pull request by dhawal1248**

### CODE REVIEW

File: v0/destinations/monetate/data/MONETATEScreen.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[
+    {
+        "destKey": "events",
+        "sourceKeys": "context.screen",
```
Comment (sayan-mitra) [inline v0/destinations/monetate/data/MONETATEScreen.json:4]: this is the screen location in mobile sdk payload?

File: v0/destinations/monetate/data/MONETATEPage.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+[
+    {
+        "destKey": "events",
+        "sourceKeys": ["properties.page", "context.page"],
+        "metadata": {
+            "targetType": "Array",
+            "isTargetTypePrimitive": false,
```
Comment (sayan-mitra) [inline v0/destinations/monetate/data/MONETATEPage.json:7]: may not require another key and find it from the "targetType"
Can discuss it in our broader refactoring plan

**Review by sayan-mitra — CHANGES_REQUESTED** [review]

File: __tests__/data/monetate_output.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        "version": "1",
+        "type": "REST",
+        "method": "POST",
+        "endpoint": "https://engine.monetate.net/api/engine/v1/decide/retailer",
+        "headers": {
+            "Content-Type": "application/json"
+        },
+        "params": {},
+        "body": "{\"monetateId\":\"1234\",\"events\":[{\"eventType\":\"monetate:context:IpAddress\",\"ipAddress\":\"0.0.0.0\"},{\"eventType\":\"monetate:context:ScreenSize\",\"height\":22,\"width\":11}],\"customerId\":\"newUser\",\"channel\":\"channel\"}",
```
[the remark this answers is not in the document]
Comment (dhawal1248) [reply __tests__/data/monetate_output.json]: Changed the type to JSON.
And also checked for key types and made necessary changes.

File: v0/destinations/monetate/data/MONETATEPage.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            "targetChildType": "object",
+            "targetFormat": {
+                "eventType": "monetate:context:PageView",
+                "url": "sourceKeyValue.url",
```
[the remark this answers is not in the document]
Comment (dhawal1248) [reply v0/destinations/monetate/data/MONETATEPage.json:11]: Yes 👍

File: v0/destinations/monetate/data/MONETATEScreen.json (hunk 1 above)
Comment (dhawal1248) [reply v0/destinations/monetate/data/MONETATEScreen.json:4]: Yes

File: __tests__/monetate.test.js [hunk 5]
```diff
@@ -17,9 +17,9 @@ const inputData = JSON.parse(inputDataFile);
 const expectedData = JSON.parse(outputDataFile);
 
 inputData.forEach((input, index) => {
-  it(`${name} Tests: payload - ${index}`, () => {
+  it(`${name} Tests: payload - ${index}`, async () => {
```
Comment (sayan-mitra) [inline __tests__/monetate.test.js:20]: why is async/await required here?
@Manashi-1881

**Review by arnab-p — APPROVED** [review]

**Pull request merged**